mesonbuild/interpreter/interpreter.py

```diff
@@ -632,8 +632,8 @@ def func_import(self, node: mparser.BaseNode, args: T.Tuple[str],
             ext_module = NotFoundExtensionModule(real_modname)
         else:
             ext_module = module.initialize(self)
-            assert isinstance(ext_module, (ExtensionModule, NewExtensionModule)), 'for mypy'
-            self.build.modules.append(real_modname)
+            assert isinstance(ext_module, (ExtensionModule, NewExtensionModule))
+            self.build.modules.add(real_modname)
```

Comment:
Is this no longer just needed for mypy?

Reply:
Gah, I pulled this out of a bigger series, and this isn't needed in *that* series because the check got moved.